src/linear_algebra/sesquilinear_form.lean

```diff
@@ -303,4 +363,215 @@ lemma is_compl_span_singleton_orthogonal {B : V →ₗ[K] V →ₗ[K] K}
 
 end orthogonal
 
+/-! ### Nondegenerate bilinear forms -/
+
+section nondegenerate
+
+section comm_semiring
+variables [comm_semiring R] [comm_semiring R₁] [add_comm_monoid M₁] [module R₁ M₁]
+  [comm_semiring R₂] [add_comm_monoid M₂] [module R₂ M₂]
+  {I₁ : R₁ →+* R} {I₂ : R₂ →+* R} {I₁' : R₁ →+* R}
+
+/-- A bilinear form is called left-separating if
+the only element that is left-orthogonal to every other element is `0`; i.e.,
+for every nonzero `x` in `M₁`, there exists `y` in `M₂` with `B x y ≠ 0`.-/
+def separating_left (B : M₁ →ₛₗ[I₁] M₂ →ₛₗ[I₂] R) : Prop :=
+∀ x : M₁, (∀ y : M₂, B x y = 0) → x = 0
+
+/-- A bilinear form is called right-separating if
+the only element that is right-orthogonal to every other element is `0`; i.e.,
+for every nonzero `y` in `M₂`, there exists `x` in `M₁` with `B x y ≠ 0`.-/
+def separating_right (B : M₁ →ₛₗ[I₁] M₂ →ₛₗ[I₂] R) : Prop :=
+∀ y : M₂, (∀ x : M₁, B x y = 0) → y = 0
+
+/-- A bilinear form is called non-degenerate if it is left-separating and right-separating. -/
+def nondegenerate (B : M₁ →ₛₗ[I₁] M₂ →ₛₗ[I₂] R) : Prop := separating_left B ∧ separating_right B
+
+lemma separating_left_flip {B : M₁ →ₛₗ[I₁] M₂ →ₛₗ[I₂] R} :
+  B.separating_left ↔ B.flip.separating_right := ⟨λ hB x hy, hB x hy, λ hB x hy, hB x hy⟩
+
+lemma separating_right_flip {B : M₁ →ₛₗ[I₁] M₂ →ₛₗ[I₂] R} :
+  separating_right B ↔ B.flip.separating_left := by rw [separating_left_flip, flip_flip]
+
+lemma separating_left_iff_neq_zero {B : M₁ →ₛₗ[I₁] M₂ →ₛₗ[I₂] R} :
+  B.separating_left ↔ ∀ x : M₁, B x = 0 → x = 0 :=
+begin
+  split; intros h x hB,
+  { let h' := h x,
+    simp only [hB, zero_apply, eq_self_iff_true, forall_const] at h',
+    exact h' },
+  have h' : B x = 0 := by { ext, rw [zero_apply], exact hB _ },
+  exact h x h',
+end
+
+lemma separating_right_iff_neq_zero_flip {B : M₁ →ₛₗ[I₁] M₂ →ₛₗ[I₂] R} :
+  B.separating_right ↔ ∀ y : M₂, B.flip y = 0 → y = 0 :=
+by rw [separating_right_flip, separating_left_iff_neq_zero]
+
+/-- A bilinear form is left-separating if and only if it has a trivial kernel. -/
+theorem separating_left_iff_ker_eq_bot {B : M₁ →ₛₗ[I₁] M₂ →ₛₗ[I₂] R} :
+  B.separating_left ↔ B.ker = ⊥ :=
+begin
+  rw linear_map.ker_eq_bot',
+  split; intro h,
+  { refine λ m hm, h _ (λ x, _),
+    rw hm, refl },
+  { intros m hm, apply h,
+    ext x, exact hm x }
+end
+
+/-- A bilinear form is right-separating if and only if its flip has a trivial kernel. -/
+theorem separating_right_iff_flip_ker_eq_bot {B : M₁ →ₛₗ[I₁] M₂ →ₛₗ[I₂] R} :
+  B.separating_right ↔ B.flip.ker = ⊥ :=
+by rw [separating_right_flip, separating_left_iff_ker_eq_bot]
+
+end comm_semiring
+
+section comm_ring
+
+variables [comm_ring R] [add_comm_group M] [module R M]
+  {I I' : R →+* R}
+
+lemma nondegenerate_of_symm_separating_left {B : M →ₗ[R] M →ₗ[R] R}
```

Comment:
```suggestion
lemma nondegenerate_of_symm_of_separating_left {B : M →ₗ[R] M →ₗ[R] R}
```
Alternative: call this `separating_left.nondegenerate_of_symm` or `is_symm.nondegenerate_of_separating_left`.
The benefit is that you can then use it with dot-notation.

Reply:
I would also expect the analogue for `separating_right`.